### PR TITLE
Create custom api client to handle authentication

### PR DESCRIPTION
**Pull request by BenWu**
This is the first step to support changes for https://bugzilla.mozilla.org/show_bug.cgi?id=1688087

This will make it easier to make changes going forward and we won't need to use (or maintain a fork) of an outdated library.

### CODE REVIEW

File: analytics_export/analyticsClient.js [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+
+  /**
+   * Retrieve account and session cookies using username and password
+   */
+  async login(username, password, testCode) {
+    const baseAuthUrl = "https://idmsa.apple.com/appleauth/auth";
+    const sessionUrl = "https://appstoreconnect.apple.com/olympus/v1/session";
+    const loginHeaders = {
+      "X-Apple-Widget-Key":
+        "e0b80c3bf78523bfe80974d320935bfa30add02e1bff88ec2166c6bd5a706c42",
```
Comment (relud) [inline analytics_export/analyticsClient.js:65]: this could use a comment explaining what this seemingly arbitrary blob is

**Review by relud — APPROVED** [review]
lgtm

Regarding 2FA, I see that is using an actual json API at `https://appstoreconnect.apple.com/analytics/api/v1`. Have you checked if it works to [use JWT auth as described for the non-analytics API](https://developer.apple.com/documentation/appstoreconnectapi) at `https://api.appstoreconnect.apple.com/v1`, rather than username/password?

**Comment by BenWu** [comment]
> Regarding 2FA, I see that is using an actual json API at `https://appstoreconnect.apple.com/analytics/api/v1`. Have you checked if it works to [use JWT auth as described for the non-analytics API](https://developer.apple.com/documentation/appstoreconnectapi) at `https://api.appstoreconnect.apple.com/v1`, rather than username/password?

I have not.  I'll try that now

**Comment by BenWu** [comment]
oh only admins can create an api key

**Comment by BenWu** [comment]
My assumption is that JWT auth won't work because all the services that offer integration require username and password but it's worth looking into and I probably should have done that sooner.  I'll move forward with this for now though

**Pull request merged**